src/bun.js/api/filesystem_router.zig

```diff
@@ -189,10 +192,25 @@ pub const FileSystemRouter = struct {
         }
 
         if (argument.get(globalThis, "dir")) |dir| {
+            if (!dir.isString()) {
+                globalThis.throwInvalidArguments("Expected dir to be a string", .{});
+                return null;
+            }
             const root_dir_path_ = dir.toSlice(globalThis, globalThis.allocator());
             if (!(root_dir_path_.len == 0 or strings.eqlComptime(root_dir_path_.slice(), "."))) {
-                root_dir_path = root_dir_path_;
+                // resolve relative path if needed
+                const path = root_dir_path_.slice();
+                if (bun.path.Platform.isAbsolute(.auto, path)) {
+                    root_dir_path = root_dir_path_;
+                } else {
+                    var parts = [_][]const u8{path};
```

Comment:
this is not necessary on canary but on debug it is, this should be removed and readDirInfo fixed